File: owl-admin/src/core/utils/storage.ts
import { AppConstants } from '../constants/appConstants';

class Storage {
  setToken(token: string): void {
    localStorage.setItem(AppConstants.tokenKey, token);
  }

  getToken(): string | null {
    return localStorage.getItem(AppConstants.tokenKey);
  }

  setUserId(userId: string): void {
    localStorage.setItem(AppConstants.userKey, userId);
  }

  getUserId(): string | null {
    return localStorage.getItem(AppConstants.userKey);
  }

  clearAuth(): void {
    localStorage.removeItem(AppConstants.tokenKey);
    localStorage.removeItem(AppConstants.userKey);
  }

  isLoggedIn(): boolean {
    const token = this.getToken();
    return token !== null && token.length > 0;
  }
}

export const storage = new Storage();

File: owl-client/lib/core/utils/storage.dart
import 'dart:convert';
import 'package:shared_preferences/shared_preferences.dart';
import '../constants/app_constants.dart';

class Storage {
  static SharedPreferences? _prefs;

  static Future<void> init() async {
    _prefs = await SharedPreferences.getInstance();
  }

  static SharedPreferences get prefs => _prefs!;

  static Future<void> setToken(String token) async {
    await prefs.setString(AppConstants.tokenKey, token);
  }

  static String? getToken() => prefs.getString(AppConstants.tokenKey);

  static Future<void> setRefreshToken(String token) async {
    await prefs.setString(AppConstants.refreshTokenKey, token);
  }

  static String? getRefreshToken() => prefs.getString(AppConstants.refreshTokenKey);

  static Future<void> setUserInfo(Map<String, String> info) async {
    await prefs.setString(AppConstants.userInfoKey, jsonEncode(info));
  }

  static Map<String, String>? getUserInfo() {
    final raw = prefs.getString(AppConstants.userInfoKey);
    if (raw == null) return null;
    try {
      return Map<String, String>.from(jsonDecode(raw));
    } catch (_) {
      return null;
    }
  }

  static Future<void> clearAuth() async {
    await prefs.remove(AppConstants.tokenKey);
    await prefs.remove(AppConstants.refreshTokenKey);
    await prefs.remove(AppConstants.userInfoKey);
  }

  static bool get isLoggedIn => getToken()?.isNotEmpty == true;
}
